x/mongo/driver/connstring/connstring.go

```diff
@@ -21,7 +21,24 @@ import (
 	"go.mongodb.org/mongo-driver/x/mongo/driver/wiremessage"
 )
 
-// Parse parses the provided uri and returns a URI object.
+// ParseAndValidate parses the provided uri and returns a URI object.
+// It check that all values are valid.
+func ParseAndValidate(s string) (ConnString, error) {
+	p := parser{dnsResolver: dns.DefaultResolver}
+	err := p.parse(s)
+	if err != nil {
+		return p.ConnString, internal.WrapErrorf(err, "error parsing uri")
+	}
+	err = p.ConnString.Validate()
+	if err != nil {
+		return p.ConnString, internal.WrapErrorf(err, "error parsing uri")
```

Comment:
Should this error message be "error validating uri" instead of "error parsing uri"?

Reply:
Yeah this was a shortcut to keep all the tests passing that expect an error that says `"error parsing uri"`. If you think it's important that we differentiate the errors, I can go through the tests and update them.

Reply:
This is likely fine as we had this error for both pieces before too.